de.marw.cmake/src/main/java/de/marw/cmake/cdt/language/settings/providers/ToolArgumentParsers.java

```diff
@@ -142,7 +142,13 @@ protected final int processArgument(List<ICLanguageSettingEntry> returnedEntries
         matcher.reset(args);
         if (matcher.lookingAt()) {
           final String name = matcher.group(oMatcher.nameGroup);
-          final String value = oMatcher.valueGroup== -1? null: matcher.group(oMatcher.valueGroup);
+          String optVal = null;
+          try {
+        	  optVal = matcher.group(oMatcher.valueGroup);          
+          } catch (IndexOutOfBoundsException e) {
```

Comment:
Please do not swallow IOOBE here, if it is thrown, then the value of `oMatcher.valueGroup` is wrong (BUG) and is reported in the log.
Swallowing the exception would remove any indication of a bug.

( I agree, I should have mentioned the exceptions in the javadoc)

Reply:
Hi, thanks for the fast reply!
I'm with you that this removes indication of bugs. So, instead of just treating the symptom (IOOBE), it is the cause which needs to be treated, which is line 363 as I tried to convey with https://regex101.com/r/isTY48/2